src/Xamarin.Android.Build.Tasks/Tasks/ProcessAssemblies.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Processes .dll files coming from @(ResolvedFileToPublish). Removes duplicate .NET assemblies by MVID.
+	/// 
+	/// Also sets some metadata:
+	/// * %(FrameworkAssembly)=True to determine if framework or user assembly
+	/// * %(HasMonoAndroidReference)=True for incremental build performance
+	/// * %(AbiDirectory) if an assembly has an architecture-specific version
+	/// </summary>
+	public class ProcessAssemblies : FilterAssemblies
```

Comment:
Not sure I see why this is a `FilterAssemblies` subclass, other than to have access to the `HasReference()` method, which could be split out "elsewhere".